routers/api/v1/user/app.go

```diff
@@ -43,8 +43,14 @@ func ListAccessTokens(ctx *context.APIContext) {
 	// responses:
 	//   "200":
 	//     "$ref": "#/responses/AccessTokenList"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
 
-	opts := auth_model.ListAccessTokensOptions{UserID: ctx.Doer.ID, ListOptions: utils.GetListOptions(ctx)}
+	if !ctx.Doer.IsAdmin && ctx.Doer != ctx.ContextUser {
+		ctx.Error(http.StatusForbidden, "", "Only admin or user self can list access tokens")
+		return
+	}
```

Comment:
This permission check can be a `reqSelf` function as there are some similar `reqXXX` functions here:
https://github.com/go-gitea/gitea/blob/e5ec57cd608a81be6e0a6b897b165e9b322ceab4/routers/api/v1/api.go#L351-L388

Reply:
Thanks very much!